server/src-lib/Hasura/GraphQL/Execute.hs

```diff
@@ -355,11 +355,13 @@ getExecPlan pgExecCtx planCache userInfo sqlGenCtx enableAL sc scVer reqUnparsed
 -- point back to them.
 rebuildFieldStrippingRemoteRels ::
      VQ.Field -> Maybe (VQ.Field, NonEmpty RemoteRelField)
+-- TODO consider passing just relevant fields (_fSelSet and _fAlias?) from Field here and modify Field in caller
 rebuildFieldStrippingRemoteRels =
   extract . flip runState mempty . rebuild 0 mempty
   where
     extract (field, remoteRelFields) =
       fmap (field, ) (NE.nonEmpty remoteRelFields)
+    -- TODO refactor for clarity
```

Comment:
This and few more (2 or 3) funcs are mind boggingly complex that definitely need refactor. Chris Done wrote them, may not have been complex for him :woman_shrugging:  :D

Reply:
Tiru to give context of this function.